src/emulator/events/types.ts

```diff
@@ -0,0 +1,86 @@
+/**
+ * The types in this file are stolen from the firebase-functions SDK. They are not subject
+ * to change because they are TS interpretations of protobuf wire formats already used in
+ * productiton services.
+ *
+ * We can't import some of them because they are marked "internal".
+ */
+
+import * as _ from "lodash";
+
+import { Resource } from "firebase-functions";
+
+/**
+ * Wire formal for v1beta1 EventFlow.
+ * Used by Firestore and some other services.
+ */
+export interface LegacyEvent {
+  data: any;
+  eventType?: string;
+  resource?: string;
+  eventId?: string;
+  timestamp?: string;
+  params?: { [option: string]: any };
+  auth?: AuthMode;
+}
+
+/**
+ * Wire format for v1beta2 Eventflow (and likely v1).
+ * Used by PubSub, RTDB, and some other services.
+ */
+export interface Event {
+  context: {
+    eventId: string;
+    timestamp: string;
+    eventType: string;
+    resource: Resource;
+  };
+  data: any;
+}
+
+/**
+ * Legacy AuthMode format.
+ */
+export interface AuthMode {
+  admin: boolean;
+  variable?: any;
+}
+
+/**
+ * Utilities for determining event types.
+ */
+export class EventUtils {
```

Comment:
I think having a file named "types" is a little odd. I would recommend (maybe in the future) to:

1. put the "types" in a `events.d.ts` file to simply declare the types
1. put the Utils functionality here in a "utils" type file to keep the code separate